src/com/nisovin/magicspells/castmodifiers/conditions/ItemHasLoreCondition.java

```diff
@@ -0,0 +1,149 @@
+
+	String typeOfCheck;
+	String loreToCheck;
+
+	@Override
+	public boolean setVar(String var) {
+		try {
+			String[] condition = var.split(";", 2);
+			typeOfCheck = condition[1];
+			loreToCheck = condition[0];
+
+			switch (typeOfCheck) {
+				case "mainhand":
+				case "offhand":
+				case "inventory":
+				case "wearing":
+					break;
+
+				default:
+					MagicSpells.error("Invalid Check-Type was specified within modifier condition.");
+					MagicSpells.error("mainhand, offhand, inventory, wearing are the only valid ones.");
+					return false;
+			}
+			return true;
+		} catch (Exception e) {
+			DebugHandler.debugGeneral(e);
+			return false;
+		}
+	}
+
+	@Override
+	public boolean check(Player player) {
+		PlayerInventory inv = player.getInventory();
+		ItemStack[] contents = inv.getContents();
+		ItemStack[] equipment = inv.getArmorContents();
+		ItemStack mainHand = HandHandler.getItemInMainHand(player);
+		ItemStack offHand = HandHandler.getItemInOffHand(player);
+
+		boolean found;
+
+		switch (typeOfCheck) {
+
+			case "mainhand":
+				found = check((ItemStack)mainHand);
+				if (found) return true;
+				break;
+
+			case "offhand":
+				found = check((ItemStack)offHand);
+				if (found) return true;
+				break;
+
+			case "inventory":
+				for (ItemStack item : contents) {
+					found = check((ItemStack)item);
+					if (found) return true;
+				}
+				break;
+
+			case "wearing":
+				for (ItemStack gear : equipment) {
+					found = check((ItemStack)gear);
+					if (found) return true;
+				}
+				break;
+		}
+		//End of Switch
+		return false;
+	}
+
+	@Override
+	public boolean check(Player player, LivingEntity target) {
+		//Wait, the target is a player? Use the Player version of check();
+		if (target instanceof Player) return check((Player)target);
+
+		EntityEquipment equip = target.getEquipment();
+		if (equip == null) return false;
+
+		ItemStack[] wearing = equip.getArmorContents();
+		ItemStack mainHand = HandHandler.getItemInMainHand(equip);
+		ItemStack offHand = HandHandler.getItemInOffHand(equip);
+
+		boolean found;
+
+		switch (typeOfCheck) {
+
+			case "mainhand":
+				found = check((ItemStack)mainHand);
+				if (found) return true;
+				break;
+
+			case "offhand":
+				found = check((ItemStack)offHand);
+				if (found) return true;
+				break;
+
+			case "wearing":
+				for (ItemStack gear : wearing) {
+					found = check((ItemStack)offHand);
+					if (found) return true;
+				}
+				break;
+			//You can't check a mob's inventory so this will return false;
+			case "inventory":
+				break;
+		}
+		return false;
+	}
+
+	@Override
+	public boolean check(Player player, Location location) {
+		return false;
+	}
+
+	private boolean check(ItemStack item) {
+		if (item == null || InventoryUtil.isNothing(item) || !item.hasItemMeta()) return false;
+
+		ItemMeta meta = item.getItemMeta();
+		if (!meta.hasLore()) return false;
+
+		List<String> itemLore = meta.getLore();
+		for (String lore : itemLore) {
```

Comment:
Is manually looping through the lines really necessary?

Reply:
List Strings do not have a suitable .contains() method in this case. Are you suggesting that I use a stream instead?

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/List.html#contains-java.lang.Object-